packages/core/src/Ponder.ts

```diff
@@ -427,5 +429,61 @@ export class Ponder {
         this.serverService.setIsHistoricalIndexingComplete();
       }
     });
+
+    // Server listeners.
+    this.serverService.on("admin:reload", async ({ chainId }) => {
+      const syncServiceForChainId = this.syncServices.find(
+        ({ network }) => network.chainId === chainId,
+      );
+      if (!syncServiceForChainId) {
+        this.common.logger.warn({
+          service: "server",
+          msg: `No network defined for chainId: ${chainId}`,
+        });
+        return;
+      }
+
+      // Clear all the metrics for the sources.
+      syncServiceForChainId.sources.forEach(({ networkName, contractName }) => {
```

Comment:
It seems like the metric resets should happen within each service. I think you could do it in the `kill` method and that would be fine? Aside, I wonder if we'll run into issues with using the same method (`kill`) for two things:
1. Shutdown service, clean up all resources, get ready for process to exit
2. Clear service internal state during a reload

Don't have an immediate answer, just a thought.

Reply:
Yeah, I think `kill()` is a bit overloaded right now. Maybe something like `stop()` when we want to do these reloads makes most sense. I'll take a look into it today.